test/validation/api/pktio/pktio.c

```diff
@@ -554,10 +554,7 @@ static int recv_packets_tmo(odp_pktio_t pktio, odp_packet_t pkt_tbl[],
 			CU_ASSERT(from_val < (unsigned)num_q);
 	} while (num_rx < num);
 
-	if (tmo == ODP_PKTIN_WAIT)
-		CU_ASSERT(num_rx == num);
-	if (num_rx < num)
-		CU_ASSERT(odp_time_to_ns(odp_time_diff(ts2, ts1)) >=  ns);
+	CU_ASSERT(num_rx == num || odp_time_diff_ns(ts2, ts1) >= ns);
```

Comment:
It's better to have separate asserts. E.g. yesterday the time diff check failed some number of times in my system. It's hard to tell what's wrong if one assert tests multiple things.